Sources/Rendering/OpenGL/BufferObject/index.d.ts

```diff
@@ -0,0 +1,87 @@
+import { ObjectType } from './Constants';
+import { vtkAlgorithm, vtkObject } from '../../../interfaces';
+
+/**
+ * Interface for initial values of BufferObject
+ */
+export interface IBufferObjectInitialValues {
+  objectType: ObjectType;
+  context?: WebGLRenderingContext | WebGL2RenderingContext;
+  allocatedGPUMemoryInBytes: number;
+}
+
+/**
+ * Base type for vtkOpenGLBufferObject excluding the 'set' method from vtkObject
+ */
+type vtkOpenGLBufferObjectBase = Omit<vtkObject, 'set'> & vtkAlgorithm;
+
+/**
+ * Interface for OpenGL Buffer Object
+ */
+export interface vtkOpenGLBufferObject extends vtkOpenGLBufferObjectBase {
```

Comment:
It looks to me that `vtkOpenGLBufferObject` merely extends `vtkObject` instead of what you have defined here.